RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractZoneIdentifier.java

```diff
@@ -0,0 +1,398 @@
+     *
+     * @return A list of PathIDs
+     *
+     * @throws TskCoreException
+     */
+    private ArrayList<Long> getPathIDsForType(BlackboardArtifact.ARTIFACT_TYPE type) throws TskCoreException {
+        ArrayList<Long> idList = new ArrayList();
+        for (BlackboardArtifact artifact : currentCase.getSleuthkitCase().getBlackboardArtifacts(type)) {
+            BlackboardAttribute pathIDAttribute = artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PATH_ID));
+
+            if (pathIDAttribute != null) {
+                long contentID = pathIDAttribute.getValueLong();
+                if (contentID != -1) {
+                    idList.add(contentID);
+                }
+            }
+        }
+        return idList;
+    }
+
+    @Messages({
+        "LOCAL_MACHINE_ZONE=Local Machine Zone",
+        "LOCAL_INTRANET_ZONE=Local Intranet Zone",
+        "TRUSTED_ZONE=Trusted Sites Zone",
+        "INTENET_ZONE=Internet Zone",
+        "RESTRICTED_ZONE=Restricted Sites Zone"
+    })
+
+    /**
+     * Wrapper class for information in the :ZoneIdentifier file. The
+     * Zone.Identifier file has a simple format of <i>key<i>=<i>value<i>. There
+     * are four known keys: ZoneId, ReferrerUrl, HostUrl, and
+     * LastWriterPackageFamilyName. Not all browsers will put all values in the
+     * file, in fact most will only supply the ZoneId. Only Edge supplies the
+     * LastWriterPackageFamilyName.
+     */
+    private final class ZoneIdentifierInfo {
+
+        private static final String ZONE_ID = "zoneid"; //NON-NLS
+        private static final String REFERRER_URL = "referrerurl"; //NON-NLS
+        private static final String HOST_URL = "hosturl"; //NON-NLS
+        private static final String FAMILY_NAME = "lastwriterpackagefamilyname"; //NON-NLS
+
+        private final HashMap<String, String> dataMap = new HashMap<>();
+
+        /**
+         * Opens the zone file, reading for the key\value pairs and puts them
+         * into a HashMap.
+         *
+         * @param zoneFile The ZoneIdentifier file
+         *
+         * @throws FileNotFoundException
+         * @throws IOException
+         */
+        ZoneIdentifierInfo(AbstractFile zoneFile) throws FileNotFoundException, IOException {
+            String line;
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(new ReadContentInputStream(zoneFile)))) {
+                while ((line = reader.readLine()) != null) {
+                    String[] tokens = line.split("=");
+
+                    if (tokens.length < 2) {
+                        continue; //Nothing interesting in this line
+                    }
+
+                    dataMap.put(tokens[0].trim().toLowerCase(), tokens[1].trim());
+                }
+            }
+        }
+
+        /**
+         * Get the integer zone id
+         *
+         * @return interger zone id or -1 if unknown
+         */
+        int getZoneId() {
```

Comment:
ZoneIdentifierInfo is a private inner class so this point is debatable, but codacy suggests all the accessor methods of ZoneIdentifierInfo should also be private.